Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabPublicHandler.cpp

```diff
@@ -706,8 +745,131 @@ namespace AzToolsFramework
 
                         for (Instance* instance : instancesInvolved)
                         {
-                            const PrefabDom& templateDom =
-                                m_prefabSystemComponentInterface->FindTemplateDom(instance->GetTemplateId());
+                            const PrefabDom& templateDom = m_prefabSystemComponentInterface->FindTemplateDom(instance->GetTemplateId());
+                            AZStd::unordered_set<AZ::IO::Path> templatePaths;
+                            PrefabDomUtils::GetTemplateSourcePaths(templateDom, templatePaths);
+
+                            if (IsCyclicalDependencyFound(afterOwningInstance->get(), templatePaths))
+                            {
+                                // Cancel the operation by restoring the previous parent
+                                AZ::TransformBus::Event(entityId, &AZ::TransformBus::Events::SetParent, beforeParentId);
+                                m_prefabUndoCache.UpdateCache(entityId);
+
+                                // Skip the creation of an undo node
+                                return AZ::Failure(AZStd::string::format(
+                                    "Reparent Prefab operation aborted - Cyclical dependency detected\n(%s depends on %s).",
+                                    instance->GetTemplateSourcePath().Native().c_str(),
+                                    afterOwningInstance->get().GetTemplateSourcePath().Native().c_str()));
+                            }
+                        }
+                    }
+                }
+
+                if (isInFocusTree && !isOwnedByFocusedPrefabInstance)
+                {
+                    // Store change as override
+                    AZ_TracePrintf("PREFAB EDIT", "OVERRIDE");
+
+                    if (isNewParentOwnedByDifferentInstance)
+                    {
+                        InternalHandleInstanceChange(parentUndoBatch, entity, beforeParentId, afterParentId);
+
+                        PrefabDom afterStateafterReparenting;
+                        m_instanceToTemplateInterface->GenerateDomForEntity(afterStateafterReparenting, *entity);
+
+                        PrefabDom newPatch;
+                        m_instanceToTemplateInterface->GeneratePatch(newPatch, afterState, afterStateafterReparenting);
+                        m_instanceToTemplateInterface->AppendEntityAliasToPatchPaths(newPatch, entityId);
+
+                        InstanceOptionalReference owningInstanceAfterReparenting =
+                            m_instanceEntityMapperInterface->FindOwningInstance(entityId);
+
+                        InternalHandleEntityOverride(
```

Comment:
It was recently pointed out by @AMZN-koppersr correctly that PrefabPublicHandler already became so big and we need to split it up. Since this is the first of many changes to come for overrides, can you create an 'overridehandler' just like the focus handler to prevent this file from growing even larger in size?

Reply:
These are existing functions that were already in the public handler, I just renamed them.
I would honestly keep things simple and move the refactoring to a follow-up PR, since we likely would not want to just move overrides stuff to its own class, but maybe the whole prefab edit workflow?